prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2462,11 +2464,21 @@ int rxr_ep_alloc_app_device_info(struct fi_info **app_device_info,
 	 * When opening a device endpoint, the info object cannot have both
 	 * FI_MR_BASIC and OFI_MR_BASIC_MAP, because FI_MR_BASIC cannot be
 	 * used with other MR modes.
-	 *
-	 * Hence unsetting the FI_MR_BASIC flag here to use the newer API
-	 * (>= 1.5).
 	 */
-	result->domain_attr->mr_mode &= ~FI_MR_BASIC;
+	if (FI_VERSION_LT(api_version, FI_VERSION(1,5))) {
```

Comment:
Suggest adding a WARN logging in both branches.

Reply:
Not sure we need to warn here. These will be used in happy path.

Reply:
I should help debug issues for customers using lower versions.

Reply:
WARN is for very few messages; asking for a specific API version is a very normal requirement. We could emit at TRACE or INFO potentially.